src/dials/algorithms/refinement/prediction/managed_predictors.py

```diff
@@ -180,7 +234,21 @@ def from_experiments(experiments, force_stills=False, spherical_relp=False):
         if force_stills:
             predictor = StillsExperimentsPredictor(experiments)
             predictor.spherical_relp_model = spherical_relp
+
         else:
-            predictor = ScansExperimentsPredictor(experiments)
+
+            all_tof_experiments = False
+            for expt in experiments:
+                if expt.scan is not None and expt.scan.has_property("time_of_flight"):
+                    all_tof_experiments = True
+                elif all_tof_experiments:
+                    raise ValueError(
+                        "Cannot find max cell for ToF and non-ToF experiments at the same time"
+                    )
```

Comment:
I'm not sure this is the right error message. This class gets called in quite a number of places, not just by the indexer where I assume the max cell issue is coming from. So we might end up here in some other situation. Maybe it should be `"Cannot create an ExperimentsPredictor for ToF and non-ToF experiments at the same time"`

Reply:
Updated the error comment.